internal/handlers/handlers.go

```diff
@@ -0,0 +1,82 @@
+package handlers
+
+import (
+	"crypto/md5"
+	"encoding/base64"
+	"fmt"
+	"github.com/go-chi/chi/v5"
+	"github.com/romanp1989/go-shortener/internal/config"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+var urlStore = make(map[string]string)
+
+func ShortenerRouter() chi.Router {
+	r := chi.NewRouter()
+
+	r.Post("/", Encode)
+	r.Get("/{id}", Decode)
+
+	return r
+}
+
+func Encode(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		http.Error(w, "Некорректный тип запроса", http.StatusBadRequest)
+	}
+
+	body, err := io.ReadAll(r.Body)
+	if err != nil || string(body) == "" {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
```

Comment:
Не стоит забывать закрывать тело ответа после чтения.
```suggestion
	body, err := io.ReadAll(r.Body)
	if err != nil || string(body) == "" {
		w.WriteHeader(http.StatusBadRequest)
		return
	}
	defer r.Body.Close()
```

Reply:
Добавил defer